apps/next/lib/types/error.ts

```diff
@@ -0,0 +1 @@
+export type SearchParamError = { message: string; status: number }
```

Comment:
Using Zod to validate the error structure we could maybe avoid creating this at all. We don't need it in the page.tsx file and if we need it in the Login component we could declare it and consume it in the file.

We might need this elsewhere in the future, but for now let's keep the file numbers to a minimum.

If we need it in the future we can always just add a directory and file for shared Zod schemas with the types using their infer utility.